### PR TITLE
Squash bugs and enhancements requested

### PR DESCRIPTION
**Pull request by kirankt**
This PR solves the following issues that were recently raised:

- Use CRD as default persistence
- Node mode now shares IPC from host
- Support node pod restarts via shared lock directory
- Add missing mount points
- Remove lvm directories by default.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -1,765 +0,0 @@
-# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
```
Comment (Akrog) [inline Gopkg.lock]: Why is this change part of this PR?  Seems unrelated.

File: pkg/controller/embercsi/embercsi_controller.go [hunk 2]
```diff
@@ -591,7 +591,6 @@ func getVolumes (ecsi *embercsiv1alpha1.EmberCSI, csiDriverMode string) []corev1
 					Path: "/var/lib/iscsi",
 				},
 			},
-			},
```
Comment (Akrog) [inline pkg/controller/embercsi/embercsi_controller.go]: This change should have been squashed with the previous commit.

File: pkg/controller/embercsi/embercsi_controller.go [hunk 3]
```diff
@@ -303,13 +306,6 @@ func getEnvVars(ecsi *embercsiv1alpha1.EmberCSI, driverMode string) []corev1.Env
 			},
 		)
 	}
-	if len(ecsi.Spec.Config.EnvVars.X_CSI_PERSISTENCE_CONFIG) > 0 {
```
Comment (Akrog) [inline pkg/controller/embercsi/embercsi_controller.go]: -1: Shouldn't we still be able to change the default configuration?  What if the admin doesn't want to use CRDs but use the DB instead?

File: pkg/controller/embercsi/embercsi_controller.go [hunk 4]
```diff
@@ -491,12 +487,8 @@ func getVolumeMounts(ecsi *embercsiv1alpha1.EmberCSI, csiDriverMode string) []co
 			Name: "iscsi-dir",
 			MountPropagation: &bidirectional,
 		},{
-			MountPath: "/etc/lvm",
```
Comment (Akrog) [inline pkg/controller/embercsi/embercsi_controller.go]: -1: This will break the LVM backend, we should add it conditionally depending on the driver being used.

File: pkg/controller/embercsi/embercsi_controller.go [hunk 5]
```diff
@@ -586,18 +585,12 @@ func getVolumes (ecsi *embercsiv1alpha1.EmberCSI, csiDriverMode string) []corev1
...
 		},{
-			Name: "lvm-dir",
+			Name: "var-lib-iscsi",
 			VolumeSource: corev1.VolumeSource{
 				HostPath: &corev1.HostPathVolumeSource{
-					Path: "/etc/lvm",
+					Path: "/var/lib/iscsi",
 				},
 			},
-		},{
```
Comment (Akrog) [inline pkg/controller/embercsi/embercsi_controller.go]: -1: This will break the LVM backend, we should add it conditionally depending on the backend.

File: pkg/controller/embercsi/embercsi_controller.go [hunk 6]
```diff
@@ -535,6 +527,13 @@ func getVolumeMounts(ecsi *embercsiv1alpha1.EmberCSI, csiDriverMode string) []co
         }
 
 	if csiDriverMode == "node" {
+		// Ember CSI shared lock directory to survive restarts
+		vm = append(vm, corev1.VolumeMount{
```
Comment (Akrog) [inline pkg/controller/embercsi/embercsi_controller.go]: -1: I don't see this being used anywhere.  Am I missing something?

**Comment by kirankt** [comment]
Ouch. Going to close this PR will add smaller ones. Thanks for the review
